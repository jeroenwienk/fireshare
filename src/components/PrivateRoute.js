import React from 'react';
import PropTypes from 'prop-types';
import Route from 'react-router-dom/Route';
import Redirect from 'react-router-dom/Redirect';

const PrivateRoute = ({ component: Component, ...rest }) => (
  <Route
    {...rest}
    render={props => {
      return rest.isAuthenticated ? (
        <Component {...props} />
      ) : (
        <Redirect
          to={{
            pathname: '/login',
            state: { from: props.location }
          }}
        />
      );
    }}
  />
);

PrivateRoute.propTypes = {
  component: PropTypes.oneOfType([PropTypes.element, PropTypes.func])
    .isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  exact: PropTypes.bool
};

export default PrivateRoute;
