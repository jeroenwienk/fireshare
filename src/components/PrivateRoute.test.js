import React from 'react';
import renderer from 'react-test-renderer';
import MemoryRouter from 'react-router-dom/MemoryRouter';
import Route from 'react-router-dom/Route';
import PrivateRoute from './PrivateRoute';

const Component = props => {
  return <div>Authenticated</div>;
};

describe('<PrivateRoute />', () => {
  test('it renders redirect when not authenticated', () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={['/']}>
          <div>
            <PrivateRoute
              exact
              component={Component}
              isAuthenticated={false}
              path="/"
            />
            <Route
              exact
              path="/login"
              render={props => {
                return <div>Unauthenticated {props.location.pathname}</div>;
              }}
            />
          </div>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders route when authenticated', () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={['/']}>
          <div>
            <PrivateRoute
              component={Component}
              isAuthenticated={true}
              path="/"
            />
          </div>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
