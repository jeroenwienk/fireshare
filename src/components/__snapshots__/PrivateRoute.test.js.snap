// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> it renders redirect when not authenticated 1`] = `
<div>
  <div>
    Unauthenticated 
    /login
  </div>
</div>
`;

exports[`<PrivateRoute /> it renders route when authenticated 1`] = `
<div>
  <div>
    Authenticated
  </div>
</div>
`;
